client/src/modules/account_statement/account_statement.service.js

```diff
@@ -0,0 +1,113 @@
+angular.module('bhima.services')
+.service('AccountStatementService', AccountStatementService);
+
+// DI
+AccountStatementService.$inject = [
+  '$uibModal', '$http', 'util',
+  'AppCache', 'FilterService', 'PeriodService',
+];
+
+/**
+ * AccountStatementService
+ */
+function AccountStatementService(Modal, $http, util, AppCache, Filters, Periods) {
+  var service = this;
+  var baseUrl = '/general_ledger/';
+  var filterCache = new AppCache('account-statement-filters');
+  var accountStatementFilters = new Filters();
+
+  // bind method
+  service.filters = accountStatementFilters;
+  service.openCommentModal = openCommentModal;
+  service.commentAccountStatement = commentAccountStatement;
+
+  /**
+   * @method commentAccountStatement
+   * @param {object} params - { uuids: An array of uuid, string: comment }
+   */
+  function commentAccountStatement(params) {
+    return $http.put(baseUrl.concat('comment'), { params : params })
+      .then(util.unwrapHttpResponse);
+  }
+
+  /**
+   * @method openCommentModal
+   * @param {object} request
+   */
+  function openCommentModal(request) {
```

Comment:
In the future, we could make this a component that would plug into every grid.  This is fine for a first implementation.